x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStore.java

```diff
@@ -200,23 +209,27 @@ private void innerPutPrivilege(ApplicationPrivilegeDescriptor privilege, WriteRe
 
     public void deletePrivileges(String application, Collection<String> names, WriteRequest.RefreshPolicy refreshPolicy,
                                  ActionListener<Map<String, List<String>>> listener) {
-        securityIndexManager.prepareIndexIfNeededThenExecute(listener::onFailure, () -> {
-            ActionListener<DeleteResponse> groupListener = new GroupedActionListener<>(
-                ActionListener.wrap(responses -> {
-                    final Map<String, List<String>> deletedNames = responses.stream()
-                        .filter(r -> r.getResult() == DocWriteResponse.Result.DELETED)
-                        .map(r -> r.getId())
-                        .map(NativePrivilegeStore::nameFromDocId)
-                        .collect(TUPLES_TO_MAP);
-                    clearRolesCache(listener, deletedNames);
-                }, listener::onFailure), names.size(), Collections.emptyList());
-            for (String name : names) {
-                ClientHelper.executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN,
-                    client.prepareDelete(SECURITY_INDEX_NAME, "doc", toDocId(application, name))
-                        .setRefreshPolicy(refreshPolicy)
-                        .request(), groupListener, client::delete);
-            }
-        });
+        if (securityIndexManager.isAvailable() == false) {
+            listener.onResponse(Collections.emptyMap());
```

Comment:
I think this should handle `exists` and `available` separately. I may have missed this one in my previous review.
I don't think we want to suggest that the delete was handled, if the index wasn't available.